build/chip/chip_test_suite.gni

```diff
@@ -111,6 +127,82 @@ template("chip_test_suite") {
+            _unique_suite_name = string_replace(string_join("-", invoker.test_sources), ".cpp", "")
+          }
+
+          silabs_executable("efr32_device_tests--${_unique_suite_name}--${_test_name}") {
+            output_name = "matter-silabs-device_tests--${_unique_suite_name}--${_test_name}.out"
+
+            defines = [ "PW_RPC_ENABLED" ]
+            sources = [
+              "${chip_root}/examples/common/pigweed/RpcService.cpp",
+              "${chip_root}/examples/common/pigweed/efr32/PigweedLoggerMutex.cpp",
+              "${chip_root}/src/test_driver/efr32/src/main.cpp",  #++++ replace absolute path
+              "${examples_common_plat_dir}/PigweedLogger.cpp",
+              "${examples_common_plat_dir}/heap_4_silabs.c",
+              "${examples_common_plat_dir}/syscalls_stubs.cpp",
+              "${examples_plat_dir}/uart.cpp",
+            ]
+
+            deps = [
+              #++++ DEBUG: Disabled code that was used for testing an alternative implementaiton.
+              # "${chip_root}/src/test_driver/efr32:nl_test_service.nanopb_rpc",  #++++ replace absolute path
+              # "${chip_root}/src/test_driver/efr32:sdk",  #++++ replace absolute path
+              #":nl_test_service--${_unique_suite_name}--${_test_name}.nanopb_rpc",
+              #":sdk--${_unique_suite_name}--${_test_name}",
+
+              "$dir_pw_unit_test:rpc_service",
+              "${chip_root}/config/efr32/lib/pw_rpc:pw_rpc",
+              "${chip_root}/examples/common/pigweed:system_rpc_server",
+              "${chip_root}/src/lib",
+              "${chip_root}/src/lib/support:pw_tests_wrapper",
+              "${chip_root}/src/lib/support:testing_nlunit",
+              "${examples_common_plat_dir}/pw_sys_io:pw_sys_io_silabs",
+              "${nlunit_test_root}:nlunit-test",
+            ]
+
+            # OpenThread Settings
+            if (chip_enable_openthread) {
+              deps += [
+                "${chip_root}/third_party/openthread:openthread",
+                "${chip_root}/third_party/openthread:openthread-platform",
+                "${examples_plat_dir}:efr-matter-shell",
+              ]
+            }
+
+            # Attestation Credentials
+            deps += [ "${examples_plat_dir}:efr32-attestation-credentials" ]
+
+            # Factory Data Provider
+            if (use_efr32_factory_data_provider) {
+              deps += [ "${examples_plat_dir}:silabs-factory-data-provider" ]
+            }
+
+            deps += pw_build_LINK_DEPS
+
+            include_dirs = [ "${chip_root}/examples/common/pigweed/efr32" ]
+
+            ldscript = "${examples_common_plat_dir}/ldscripts/${silabs_family}.ld"
+
+            inputs = [ ldscript ]
+
+            ldflags = [
+              "-T" + rebase_path(ldscript, root_build_dir),
+              "-Wl,--no-warn-rwx-segment",
+            ]
+
+            output_dir = root_out_dir
+          }
+        }
+
         pw_test(_test_name) {
```

Comment:
The PW version of `chip_test_suite` template makes calls to `pw_test` (part of pigweed) for each test source instead of calling `chip_test` (part of our codebase).  Currently `chip_test` is only being used by `chip_test_suite_using_nltest`, so I presume it has not been updated to work with PW unit testing.  If `chip_test` were updated, we could make it so that `chip_test_suite` calls `chip_test` (instead of `pw_test`), and then I could probably move all of the `silabs_executable` stuff and the efr32 stuff into `chip_test` rather than `chip_test_suite`.